pkg/apis/pipeline/v1/pipeline_validation_test.go

```diff
@@ -2599,6 +2648,97 @@ func TestContextInvalid(t *testing.T) {
 			Also(apis.ErrGeneric(`non-existent variable in "$(context.pipeline.missing-foo)"`, "value")).
 			Also(apis.ErrGeneric(`non-existent variable in "$(context.pipelineRun.missing-foo)"`, "value")).
 			Also(apis.ErrGeneric(`non-existent variable in "$(context.pipelineTask.missing-foo)"`, "value")),
+	}, {
+		name: "invalid string context variable for pipeline in include matrix",
+		tasks: []PipelineTask{{
+			Name:    "bar",
+			TaskRef: &TaskRef{Name: "bar-task"},
+			Params: []Param{{
+				Name: "a-param", Value: ParamValue{StringVal: "$(context.pipeline.missing)"},
```

Comment:
it looks like this tests an invalid param in both task.params and matrix.include.params; can it be scoped down to testing matrix.include only?

Reply:
It doesn't look like this comment was addressed, can you please take another look here? this applies to all the test cases

Reply:
Sorry I was thinking it was the Matrix.Params you were referring to instead of the Task.Params. This has been updated